src/app/services/data/trainrunsection.service.ts

```diff
@@ -879,18 +879,30 @@ export class TrainrunSectionService implements OnDestroy {
     let summedTravelTime = 0;
 
     const iterator = this.trainrunService.getNonStopIterator(leftNode, trs);
+    let prevInitialTargetArrival: number = null;
+    let stopTime: number;
     while (iterator.hasNext()) {
       const nextPair = iterator.next();
 
-      const isRightNodeNonStop = nextPair.node.isNonStop(nextPair.trainrunSection);
-      trsTimeStructure.travelTime = TrainrunsectionHelper.getTravelTime(
+      if (prevInitialTargetArrival !== null) {
+        stopTime = MathUtils.mod60(nextPair.trainrunSection.getSourceDeparture() - prevInitialTargetArrival);
+        trsTimeStructure.leftDepartureTime = trsTimeStructure.rightArrivalTime + stopTime;
+        trsTimeStructure.leftArrivalTime = trsTimeStructure.rightDepartureTime + stopTime;
```

Comment:
```suggestion
        trsTimeStructure.leftArrivalTime = trsTimeStructure.rightDepartureTime - stopTime;
```

Since the train goes from: 
- `left -> right` then 
- `left <- right`

You can try on a non-stop chain with more than 1 collapsed node

[reticulaire(3).json](https://github.com/user-attachments/files/23303826/reticulaire.3.json)